source/codegen/templates/service_helpers.mako

```diff
@@ -208,6 +210,14 @@ ${initialize_standard_input_param(function_name, parameter)}\
         ${parameter_name}_request.end(),
         std::back_inserter(${parameter_name}),
         [](auto x) { return x ? VI_TRUE : VI_FALSE; });
+ % elif common_helpers.is_struct(parameter) and common_helpers.is_array(c_type):
+      auto ${parameter_name}_request = ${request_snippet};
+      std::vector<${c_type_underlying_type}> ${parameter_name};
+      std::transform(
+        ${parameter_name}_request.begin(),
+        ${parameter_name}_request.end(),
+        std::back_inserter(${parameter_name}),
+        [&](${namespace_prefix}${grpc_underlying_type} x) { return GetStructFromGrpcType(x); }); \
```

Comment:
Since the function has the signature expected by `std::transform`, you don't need to wrap it in a lambda:
```cpp
std::transform(
  ${parameter_name}_request.begin(),
  ${parameter_name}_request.end(),
  std::back_inserter(${parameter_name}),
  GetStructFromGrpcType);
```

Reply:
If I dont use the lambda, I get an error saying 'No matching overload found'
![image](https://user-images.githubusercontent.com/47028785/120822697-50f45480-c574-11eb-8fa1-e01f0be0ad0f.png)

Reply:
If there are multiple `GetStructFromGrpcType` functions for different struct types, just specifying function name fails to resolve to correct overload. I think that manipulates in above compile time error.

Reply:
I have left the lambda as is.